src/components/services/pathutil.ts

```diff
@@ -1,80 +1,110 @@
-        var fragments = PathUtil.toPropertyFragments(schemaPath);
-        return '/' + fragments.slice(0, fragments.length - 1).join('/');
-    };
-
-    static filterIndexes(path:string):string {
-        return PathUtil.toPropertyFragments(path).filter(function (fragment, index, fragments) {
-            return !(fragment.match("^[0-9]+$") && fragments[index - 1] == "items");
-        }).join("/");
+        let fragments = PathUtil.toPropertyFragments(propertyPath);
+        return fragments.reduce((propertyAccessString, fragment) =>
+            `${propertyAccessString}['${fragment}']`
+            , "");
     }
 
-    static filterNonKeywords = (fragments:string[]):string[] => {
-        return fragments.filter(function (fragment) {
-            return !(PathUtil.Keywords.indexOf(fragment) !== -1);
-        });
-    };
 
-    static beautifiedLastFragment(schemaPath: string): string  {
-        return PathUtil.beautify(PathUtil.capitalizeFirstLetter(PathUtil.lastFragment(schemaPath)));
+    /**
+     * Gets all but the last fragments of the given path as a string.
+     *
+     * @param schemaPath the path from which to retrieve all but the last fragment
+     * @returns {string} the path without the last fragment
+     */
+    static init(schemaPath: string): string {
+        return '/' + _.flow(PathUtil.toPropertyFragments, _.initial, PathUtil.joinWithSlash)(schemaPath);
     }
 
-    static lastFragment(schemaPath: string): string {
-        return schemaPath.substr(schemaPath.lastIndexOf('/') + 1, schemaPath.length)
+    /**
+     * Removes all array indices from an instance path
+     *
+     * @param path the instance path from which to remove indices
+     * @returns {string} the filtered path without indices
+     */
+    static filterIndexes(path:string):string {
+        return PathUtil.toPropertyFragments(path)
+            .filter((fragment, index, fragments) =>
+                !(fragment.match(PathUtil.numberRegex) && fragments[index - 1] == "items")
+            ).join("/");
     }
 
-    private static capitalizeFirstLetter(string): string {
-        return string.charAt(0).toUpperCase() + string.slice(1);
+    /**
+     * Removes all schema-specific fragments from the given path.
+     *
+     * @param fragments the path from which to filter out the schema specific fragment
+     * @returns {string[]} a path without schema specific fragments
+     */
+    static filterNonKeywords(fragments:string[]):string[] {
+        return fragments.filter(fragment => !_.includes(PathUtil.Keywords, fragment))
     }
 
+    /**
+     * Returns the last fragment of the given path in a beautified manner.
+     *
+     * @param schemaPath the path from which to extract the beautified fragment
+     * @returns {any} the last fragment in a beautified manner
+     */
+    static beautifiedLastFragment(schemaPath: string): string  {
+        return _.flow(PathUtil.lastFragment, _.upperFirst, PathUtil.beautify)(schemaPath);
```

Comment:
the _.uperFirst call is not necessary as the beautify method will upper case the string